### PR TITLE
[4] PHP 8.2 Dynamic property - registeredurlparams- cache enabled

### PR DESCRIPTION
**Pull request by alikon**
### Summary of Changes
declared `$registeredurlparams`

### Testing Instructions
open frontend with php 8.2 with Error Reporting = maximum and cache enabled

### Actual result BEFORE applying this Pull Request
Deprecated: Creation of dynamic property Joomla\CMS\Application\SiteApplication::$registeredurlparams is deprecated

### Expected result AFTER applying this Pull Request
no depraction

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: libraries/src/Application/SiteApplication.php [hunk 1]
```diff
@@ -56,6 +56,14 @@ final class SiteApplication extends CMSApplication
      * @since  4.0.0
      */
     protected $detect_browser = false;
+ 
+    /**
```
Comment (richard67) [inline libraries/src/Application/SiteApplication.php]:
```suggestion

    /**
```
Fix PHPCS error "Whitespace found at end of line" for line 59.

**Comment by carlitorweb** [comment]
I have tested this item :white_check_mark: successfully on bc242fcc66b6f44da74466710e17d23b997d40ac<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39683">issues.joomla.org/tracker/joomla-cms/39683</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on bc242fcc66b6f44da74466710e17d23b997d40ac<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39683">issues.joomla.org/tracker/joomla-cms/39683</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39683">issues.joomla.org/tracker/joomla-cms/39683</a>.</sub>

**Comment by fancyFranci** [comment]
Why the "needs deprecation" @laoneo ?

**Comment by laoneo** [comment]
Because this variable is not used in core and I don't see a reason for it.

**Comment by joomdonation** [comment]
> Because this variable is not used in core and I don't see a reason for it.

@laoneo Actually, we are using it in core. See https://github.com/joomla/joomla-cms/blob/4.2-dev/libraries/src/MVC/Controller/BaseController.php#L613-L629 and https://github.com/joomla/joomla-cms/blob/4.2-dev/libraries/src/Cache/Cache.php#L738

Look like we are using that variable to pass data from controller to cache. So until we find a better way to pass that data, it should not be deprecated.

**Comment by laoneo** [comment]
Somehow I missed as I was searching for this string. Anyway, this is the wrong approach and should be done in a different way as the app object should only have member variables it actually uses and should not act as a basket for any kind if in memory storage data.

**Comment by fancyFranci** [comment]
That's right, but out of scope of this PR. At least the wrong approach is more visible now.

**Comment by laoneo** [comment]
The needs deprecation label is used to identify prs which need some followup prs for 4.3 or 5.0 with the introduction of deprecations and should not be removed. Please leve them in place @fancyFranci as I was explaining this already un maintainers chat.

**Comment by fancyFranci** [comment]
Then I misunderstood your last answer that it should be refactored, which is in my opinion not the same as setting a deprecation label, but ok.

**Comment by fancyFranci** [comment]
Thank you for fixing that!

**Pull request merged**